frontend/public/components/edit-yaml.jsx

```diff
@@ -235,12 +235,26 @@ export const EditYAML_ = connect(stateToProps)(
           });
         }
 
+        appendYAMLString(yaml) {
+          const currentYAML = this.getEditor().getValue();
+          return _.isEmpty(currentYAML)
+            ? yaml
+            : `${currentYAML.trim()}${
+                currentYAML.trim().endsWith('---') ? '\n' : '\n---\n'
```

Comment:
We should avoid trimming if possible. That could change the actual content. It's an edge case, but the last property might require trailing whitespace.